tests/std/tests/P0896R4_ranges_algorithm_machinery/test.cpp

```diff
@@ -196,15 +195,28 @@ namespace indirect_binary_predicate_test {
         std::false_type operator()(simple_common_reference, simple_common_reference) const requires(I != 5);
     };
 
-    STATIC_ASSERT(!indirect_binary_predicate<Fn<0>, simple_iter_archetype<1>, simple_iter_archetype<1>>);
-    STATIC_ASSERT(!indirect_binary_predicate<Fn<1>, simple_iter_archetype<1>, simple_iter_archetype<1>>);
-    STATIC_ASSERT(!indirect_binary_predicate<Fn<2>, simple_iter_archetype<1>, simple_iter_archetype<1>>);
-    STATIC_ASSERT(!indirect_binary_predicate<Fn<3>, simple_iter_archetype<1>, simple_iter_archetype<1>>);
-    STATIC_ASSERT(!indirect_binary_predicate<Fn<4>, simple_iter_archetype<1>, simple_iter_archetype<1>>);
-    STATIC_ASSERT(!indirect_binary_predicate<Fn<5>, simple_iter_archetype<1>, simple_iter_archetype<1>>);
-    STATIC_ASSERT(!indirect_binary_predicate<Fn<6>, simple_iter_archetype<0>, simple_iter_archetype<1>>);
-    STATIC_ASSERT(!indirect_binary_predicate<Fn<6>, simple_iter_archetype<1>, simple_iter_archetype<0>>);
-    STATIC_ASSERT(indirect_binary_predicate<Fn<6>, simple_iter_archetype<1>, simple_iter_archetype<1>>);
+    template <int FuncSelector, int IterSelector1, int IterSelector2>
+    constexpr bool test() {
+        using std::indirect_binary_predicate, std::indirect_equivalence_relation, std::indirect_strict_weak_order;
+        using F               = Fn<FuncSelector>;
+        using I1              = simple_iter_archetype<IterSelector1>;
+        using I2              = simple_iter_archetype<IterSelector2>;
+        constexpr bool result = indirect_binary_predicate<F, I1, I2>;

```

Comment:
Style: Perhaps a newline between the `using` and the `bool result` would make things look nicer? Feel free to disregard.